reconcile/utils/qontract_server_client.py

```diff
@@ -0,0 +1,95 @@
+from urllib.parse import urlparse
+
+import requests
+
+
+class QontractServerClient:
+    """REST API for Qontract Server """
+
+    def __init__(self, url, token=None,
+                 use_sessions=False,
+                 sha_url=False):
+        self.base_url = url
+        self.token = token
+        self.sha = None
+
+        if use_sessions:
+            self.session = requests.Session()
+        else:
+            self.session = None
+
+        if sha_url:
+            self.set_graphqlsha_url()
+        else:
+            self.query_url = url
+
+    def set_graphqlsha_url(self):
+        """Fetches the latest sha and sets the query url to use it.
+        """
+        self.sha = self.get_sha()
```

Comment:
It's a personal choice I guess, but I always find the class variables initialization outside of the `__init__` confusing. My suggestion is to return values from the helpers and to have all the initializations done on the `__init__`.